python/lsst/obs/base/script/convert.py

```diff
@@ -72,11 +71,20 @@ def convert(repo, gen2root, instrument, skymap_name, skymap_config, calibs, reru
         butlerConfig = repo
 
     butler = lsst.daf.butler.Butler(butlerConfig)
-    instr = getInstrument(instrument, butler.registry)
+
+    # Derive the gen3 instrument from the gen2root
+    # This requires we instantiate a gen2 butler solely to get its mapper
+    with temporaryLogLevel("CameraMapper", Log.ERROR):
+        with temporaryLogLevel("HscMapper", Log.ERROR):
```

Comment:
This seems kind of specific.  Is this the only mapper logger that doesn't use `CameraMapper`?

Reply:
I was wondering about that. It's the code that is in the other bit of the conversion code that I lifted it from. It seemed like @TallJimbo needed that in one place so I'm going to need it here. It's a shame the mappers don't share a log context.  obs_lsst uses `"LsstCamMapper"` so I probably need to suppress that as well (since it looks like it warns in the constructor). Or we remove HscMapper and live with the irrelevant chatter.

Reply:
I think I should just do this for the root mapper instead and hide everything. The system will issue the warnings a bit later anyhow.  Talking of which, on my laptop it takes 10 seconds to load the imsim mapper and 14 seconds for the ts3 mapper. It's a shame there's no way to short circuit the yaml camera parsing or defer it until we need it. (80% of the time is spent calling _makeCamera).

Reply:
I think I can make the time penalty disappear if I make this change to obs_lsst:

```diff
diff --git a/python/lsst/obs/lsst/lsstCamMapper.py b/python/lsst/obs/lsst/lsstCamMapper.py
index 9db3744..44e433b 100644
--- a/python/lsst/obs/lsst/lsstCamMapper.py
+++ b/python/lsst/obs/lsst/lsstCamMapper.py
@@ -23,6 +23,7 @@
 """The LsstCam Mapper."""  # necessary to suppress D100 flake8 warning.
 
 import os
+from functools import lru_cache
 import lsst.log
 import lsst.geom
 import lsst.utils as utils
@@ -166,7 +167,13 @@ class LsstCamBaseMapper(CameraMapper):
         camera : `lsst.afw.cameraGeom.Camera`
             Camera geometry.
         """
+        return cls._makeYamlCamera(cameraYamlFile=cameraYamlFile)
 
+    @classmethod
+    @lru_cache(maxsize=10)
+    def _makeYamlCamera(cls, cameraYamlFile=None):
+        """Helper function for _makeCamera that can be cached.
+        """
         if not cameraYamlFile:
             cameraYamlFile = os.path.join(utils.getPackageDir(cls.packageName), "policy",
                                           ("%s.yaml" % cls.getCameraName()))
~
```
The new method has to exist because policy is not hashable but it's not used anyhow. This means I can instantiate later mappers almost instantly.